cpp/nanobind/nanobind.cc

```diff
@@ -39,6 +42,32 @@ std::vector<std::string> CommonEncodedVocabType(
   return encoded_vocab_strs;
 }
 
+bool GrammarMatcher_FillNextTokenBitmask(
+    GrammarMatcher& matcher,
+    nb::ndarray<int32_t, nb::device::cpu> arr,
```

Comment:
Can we accept any array here and report a customized error msg when the array is not on CPU? The msg can be
```
fill_next_token_bitmask requires the tensor to be on CPU, but got device ...
```

Besides, the difference between nanobind.cc and python_method.h/.cc is: the former is just binding, and the latter defines helpful functions to bind. So it's still good to put the methods in python_methods.h/.cc (and include nanobind.h there). Of course this is just a matter of code structure and doesn’t affect whether this PR will be merged. But we can eventually move these methods there.